bin-prot/src/value/mod.rs

```diff
@@ -26,11 +26,11 @@ pub enum Value {
     Nat0(u32),
     Bool(bool),
     String(Vec<u8>),
-    Char(char),
+    Char(u8),
     Int(i64),
     Float(f64),
     Option(Option<Box<Value>>),
-    Record(HashMap<String, Value>), // records/structs
+    Record(Vec<(String, Value)>), // records/structs
```

Comment:
Important lesson here. Never use `HashMap` for something that needs deterministic ordering of fields. This led to flaky failing tests because the ordering would randomly change

Reply:
or can use https://docs.rs/indexmap/1.7.0/indexmap/ too